### PR TITLE
r2.2 systemlibs cherry-picks

### PR DESCRIPTION
**Pull request by perfinion**
These are cherry-picks for the r2.2 release branch. It fixes several issues for building against system libraries. I've pushed these patches to the Gentoo package already so they have some testing but it'd be good to get them merged for the release as well for other distros too.

The commits are in master in PRs:
https://github.com/tensorflow/tensorflow/pull/38326
https://github.com/tensorflow/tensorflow/pull/38327
https://github.com/tensorflow/tensorflow/pull/38328

@angerson @mihaimaruseac

### CODE REVIEW

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by goldiegadde** [comment]
@perfinion @mihaimaruseac presubmits are failing, could you take a look ?

**Comment by mihaimaruseac** [comment]
Yes, the pybind part of the cherry-pick needs slightly more work. Let's wait until we can merge the 3 PRs into master

**Comment by perfinion** [comment]
@goldiegadde I dropped the jsoncpp patch since im having issues getting kokoro to pass. the other patches are in master now :)

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**
